From 5db9ce2e8ebcdac22c46802d28e53e91b5ca3d6f Mon Sep 17 00:00:00 2001
From: Ramya Gnanasekar <quic_rgnanase@quicinc.com>
Date: Thu, 30 Nov 2023 16:36:38 +0530
Subject: [PATCH] wpa_supplicant: Add channel 140 to htplus allowed list

When channel 140 is configured in mesh, interface fails to come up
due to channel bond (136,140). Channel 136 is not HT40+ capable and hence
validation for HT channel bonding fails.

In mesh, during channel setup, secondary channel offset for the
configured channel will be selected as +1 if primary channel is capable of HT40+.
In current code, channel 140 is not allowed as HT40+ and hence secondary
channel offset is selected as -1, 136 is selected as secondary channel.
But channel 136 is not HT40+ supported and fails in channel bonding
validation.

Fix this by adding 140 to HT40+ allowed list because it is capable of
HT40+

Signed-off-by: Ramya Gnanasekar <quic_rgnanase@quicinc.com>
---
 wpa_supplicant/wpa_supplicant.c | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/wpa_supplicant/wpa_supplicant.c b/wpa_supplicant/wpa_supplicant.c
index 0c05c2b..a44db0e 100644
--- a/wpa_supplicant/wpa_supplicant.c
+++ b/wpa_supplicant/wpa_supplicant.c
@@ -2789,8 +2789,8 @@ void ibss_mesh_setup_freq(struct wpa_supplicant *wpa_s,
 	int ieee80211_mode = wpas_mode_to_ieee80211_mode(ssid->mode);
 	enum hostapd_hw_mode hw_mode;
 	struct hostapd_hw_modes *mode = NULL;
-	int ht40plus[] = { 36, 44, 52, 60, 100, 108, 116, 124, 132, 149, 157,
-			   165, 173, 184, 192 };
+	int ht40plus[] = { 36, 44, 52, 60, 100, 108, 116, 124, 132, 140, 149,
+			   157, 165, 173, 184, 192 };
 	/* bw_80_160 array members are 80MHz start freq, 80MHz end freq and so on
 	 */
 	unsigned int bw_80_160[] = { 5180, 5240, 5260, 5320,
-- 
2.17.1

